aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -280,16 +283,16 @@ struct cauchy_distribution {
 template <typename T>
 struct lognormal_distribution {
 
-  inline lognormal_distribution(T mean_in, T stdv_in) {
-    TORCH_CHECK(stdv_in > 0);
+  C10_HOST_DEVICE inline lognormal_distribution(T mean_in, T stdv_in) {
+    TORCH_CHECK_IF_NOT_ON_CUDA(stdv_in > 0);
     mean = mean_in;
     stdv = stdv_in;
   }
 
   template<typename RNG>
-  inline T operator()(RNG generator){
-    normal_distribution<T> normal(mean, stdv);
-    return ::exp(normal(generator));
+  C10_HOST_DEVICE inline T operator()(RNG generator){
+    normal_distribution<T> normal(0.0, 1.0);
```

Comment:
This seems like a pessimization. Look at the godbolt code here: https://godbolt.org/z/Qt2RVo You can see that the compiler refuses to optimize away the multiplication by 1.0 and addition by 0.0, even though "mathematically" they are no-ops. I have to past `-ffast-math` for the optimization to take place. We don't compile PyTorch with fast-math.

Reply:
addressed